src/ContinuosBondingERC20Token.sol

```diff
@@ -201,45 +208,54 @@ contract ContinuosBondingERC20Token is ERC20, ReentrancyGuard {
         currentEth = currentEth > address(this).balance ? address(this).balance : currentEth;
         isLpCreated = true;
 
-        //is this line necessary? can be removed
-        // _approve(address(this), address(router), currentTokenBalance);
-
-        address wNative;
-        uint256 liquidity;
-        if (poolType == LP_POOL.Uniswap) {
-            wNative = IUniswapV2Router02(router).WETH();
-
-            address pair = factory.getPair(address(this), wNative);
-            if (pair == address(0)) pair = factory.createPair(address(this), wNative);
-            IWETH(wNative).deposit{ value: currentEth }();
-            IWETH(wNative).transfer(pair, currentEth);
-            _transfer(address(this), pair, currentTokenBalance);
-            liquidity = IPair(pair).mint(address(this));
-        } else if (poolType == LP_POOL.TraderJoe) {
-            wNative = IJoeRouter02(router).WAVAX();
-
-            address pair = factory.getPair(address(this), wNative);
-            if (pair == address(0)) pair = factory.createPair(address(this), wNative);
-            IWETH(wNative).deposit{ value: currentEth }();
-            IWETH(wNative).transfer(pair, currentEth);
-            _transfer(address(this), pair, currentTokenBalance);
-            liquidity = IPair(pair).mint(address(this));
+        // Create the pool if it doesn't exist
+        (address token0, address token1) = address(this) < WETH ? (address(this), WETH) : (WETH, address(this));
+        address pool = nonfungiblePositionManager.createAndInitializePoolIfNecessary(token0, token1, 30000, _getSqrtPriceX96(currentEth, currentTokenBalance));
```

Comment:
> `_getSqrtPriceX96(currentEth, currentTokenBalance))`
the `sqrtPrice` stored in uniswap v3 is in terms of amount  of token1/ amount of token0. So, it will change in terms of token0 and token1 determined in earlier step.